rust/src/proofs/api.rs

```diff
@@ -174,6 +192,42 @@ fn seal_pre_commit_phase2(
     })
 }
 
+/// TODO: document
+#[ffi_export]
+fn generate_tree_r_last(
+    registered_proof: RegisteredSealProof,
+    replica_path: c_slice::Ref<u8>,
+    output_dir: c_slice::Ref<u8>,
+) -> repr_c::Box<GenerateTreeRLastResponse> {
+    catch_panic_response("generate_tree_r_last", || {
+        let comm_r_last = seal::generate_tree_r_last(
+            registered_proof.into(),
+            as_path_buf(&replica_path)?,
+            as_path_buf(&output_dir)?,
+        )?;
+
+        Ok(comm_r_last)
+    })
+}
+
+/// TODO: document
+#[ffi_export]
+fn generate_tree_c(
+    registered_proof: RegisteredSealProof,
+    input_dir: c_slice::Ref<u8>,
+    output_dir: c_slice::Ref<u8>,
```

Comment:
Are both of those the "cache directory"? I'm assuming it's fine if we set both to the same path.

Reply:
Yes, in your case they would both point to the same "cache directory". I just wanted to propagate the flexibility to have different ones upwards, in case it's needed in the future.